src/AstToCFlat.ml

```diff
@@ -899,32 +899,57 @@ and mk_expr (env: env) (locals: locals) (e: expr): locals * CF.expr =
 
 (* See digression for [dup32] in CFlatToWasm *)
 let scratch_locals =
-  [ I64; I64; I32; I32 ]
+  [ I64; I64; I32; I32; I32 ]
 
 let msg_i64 = format_of_string "Abort: %s was meant to be hand-written and \
   provided at link-time, but contains an I64 and therefore cannot be called from \
   WASM."
 
-let mk_decl env (d: decl): env * CF.decl option =
+let mk_wrapper orig_name n_args locals =
```

Comment:
the strategy for returning 64-bit integers to javascript is as follows:
- if the function `f` is part of the public API
- and returns a uint64
then we generate a _packed variant that calls `f`, then decomposes the uint64 return value as a pair of uint32s

since krml does not take advantage of multi-value returns (it should! but these weren't supported when I wrote the backend), a special instruction here allows converting a 64-bit integer into two 32-bit integers...

in the future, it'd be good to add general support for multi-values, and then simply call WasmSupport.u64tou32s of type `uint64 -> uint32 & uint32`